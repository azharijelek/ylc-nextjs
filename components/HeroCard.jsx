export default function HeroCard(props) {
    return (
        <>
            <section className="hs-card">
                <h2>{props.title}</h2>
                <img src={props.thumbnail} alt={props.title} loading="lazy"/>
            </section>
            
            <style jsx>{`
            .hs-card {
                width: 300px;
                height: 200px;
                background: #ddd;
                border-radius: 8px;
                position: relative;
                overflow: hidden;
                background: #f5f5f5;

                h2 {
                    display: block;
                    width: 100%;
                    height: auto;
                    background: rgba(0,0,0,.7);
                    padding: 13px;
                    font-size:16px;
                    font-weight: bold;
                    position: absolute;
                    bottom: 0;
                    left: 0;
                    color: #fff;
                    margin: 0;
                    text-align:left;
                    min-height: 70px;
                }

                img {
                    width: 100%;
                    height: 100%;
                    object-fit: cover;
                }
            }
            `}</style>
        </>
    )
}