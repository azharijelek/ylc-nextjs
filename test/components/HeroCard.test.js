import React from 'react'
import Adapter from 'enzyme-adapter-react-16'
import { shallow, configure } from 'enzyme'
import HeroCard from '@/components/HeroCard'

configure({ adapter: new Adapter() })

describe('<HeroCard/>', () => {
  const mainProps = {
    thumbnail: '',
    title: '',
    permalink: ''
  }

  it('renders correctly', () => {
    const wrapper = shallow(<HeroCard />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should pass the props', () => {
    const img_url =
      'https://cdn.statically.io/img/yourlifechoices.s3-ap-southeast-2.amazonaws.com/old/assets/autouploads/d62a2631cf50.jpg?f=webp'
    const permalink =
      'http://localhost:4200/travel/travel-resources/resources-travel-experiences/ten-things-to-stop-on-aeroplanes'
    const title = 'Lorem ipsum'

    const wrapper = shallow(
      <HeroCard {...mainProps} thumbnail={img_url} title={title} permalink={permalink} />
    )

    expect(wrapper.find('img').prop('src')).toBe(img_url + '&h=200')
    expect(wrapper.find('h2').find('a').prop('title')).toBe(title)
    expect(wrapper.find('h2').find('Link').prop('href')).toBe(permalink)
  })
})
