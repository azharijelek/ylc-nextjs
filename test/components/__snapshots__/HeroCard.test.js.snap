// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroCard/> renders correctly 1`] = `ShallowWrapper {}`;
